partiql-ast/src/main/java/org/partiql/ast/Identifier.java

```diff
@@ -35,11 +39,49 @@ public <R, C> R accept(@NotNull AstVisitor<R, C> visitor, C ctx) {
     }
 
     @NotNull
-    public String getSymbol() {
-        return this.symbol;
+    public List<Part> getQualifier() {
+        return this.qualifier;
     }
 
-    public boolean isDelimited() {
-        return this.delimited;
+    @NotNull
+    public Part getBase() {
+        return this.base;
+    }
+
+    /**
+     * Represents a part of a SQL identifier, such as {@code c} in {@code a.b.c}.
+     */
+    @lombok.Builder(builderClassName = "Builder")
+    @EqualsAndHashCode(callSuper = false)
+    public static final class Part extends AstNode {
+        @NotNull
+        private final String symbol;
+
+        private final boolean delimited;
+
+        public Part(@NotNull String symbol, boolean delimited) {
+            this.symbol = symbol;
+            this.delimited = delimited;
+        }
+
+        @NotNull
+        @Override
+        public List<AstNode> getChildren() {
+            return new ArrayList<>();
+        }
+
+        @Override
+        public <R, C> R accept(@NotNull AstVisitor<R, C> visitor, C ctx) {
+            return visitor.visitIdentifierPart(this, ctx);
+        }
+
+        @NotNull
+        public String getSymbol() {
+            return this.symbol;
+        }
+
+        public boolean isDelimited() {
+            return this.delimited;
+        }
     }
```

Comment:
Did you consider adding the same helper methods that SPI has for creating identifiers? It is cumbersome to create them, but the helper methods are an easy addition with a lot of utility.

Reply:
Yeah, I was planning to add them. I didn't get a chance to add them yesterday